src/adaptors/multi_product.rs

```diff
@@ -122,10 +123,11 @@ where
             // Only the first time.
             None => {
```

Comment:
Analogously, would it make sense to replace this `Option` by `enum CurrentValues { NotYetPopulated, Populated(Vec<_>) } `?